### ISSUE TITLE
[BUG] CloneSet clientset API Create method changes the cloneset object to create

### ISSUE THREAD

**Issue by wgitg** [openkruise/kruise#373]
I have a Foo CRD and controller, the controller is creating a cloneset object based on what's specified in a Foo object.
The foo controller is using code-generator's clientset, lister and informer. 
I use openkruise/kruise, where there are generated lister/clientset/informers as well, see the import packages I used.
```
        kruiseappsv1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
	kruisev1alpha1clientset "github.com/openkruise/kruise/pkg/client/clientset/versioned"
	kruisev1alpha1informer "github.com/openkruise/kruise/pkg/client/informers/externalversions/apps/v1alpha1"
	kruisev1alpha1listers "github.com/openkruise/kruise/pkg/client/listers/apps/v1alpha1"
```
I used the those packages to do `syncHandler` similarly as 
sample-[controller logic](https://github.com/kubernetes/sample-controller/blob/master/controller.go)

basically we first try to calling lister
	`cloneset, err = c.cloneSetLister.CloneSets(foo.Namespace).Get(clonesetBuilder.GetIndexKey())`
If it does not exist, then we create a new cloneset and get the returned value.
```         
                      // fooCloneSetSpec is created based on Foo spec
                      fooCloneSet := clonesetBuilder.NewOrUpdateCloneSet(fooCloneSetSpec)
			klog.Infof("Before calling to the API, cloneset Name: %v", fooCloneSet.Name) 
			cloneset, err = c.cloneSetClientset.AppsV1alpha1().CloneSets(foo.Namespace).Create(
				fooCloneSet,
			)
                      klog.Infof("After calling into the API, cloneset Name: %v", cloneset.Name) 
```

**What happened**:
I started my controller, and started openkruise controller (with only cloneset controller installed and running)

I see the printed out the message: 

Before calling to the API, cloneset Name: "foo-cloneset-1"
After calling into the API, cloneset Name: ""

The ownerReference is also "resetted", where `metav1.IsControlledBy(cloneset, foo)` returns false
**What you expected to happen**:

Before calling to the API, cloneset Name: "foo-cloneset-1"
After calling into the API, cloneset Name: "foo-cloneset-1"
And the ownerReference calls `metav1.IsControlledBy(cloneset, foo)` should return true

The cloneset controller is written using kubebuilder generated controller flow, and I am not sure that whether cloneset can work with the generated clientset/lister/informer? But even that, I'd expect that the cloneset controller uses  informer/clientset indirectly from the "controller-runtime" library...
 
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kruise version: v0.6.0
- Kubernetes version (use `kubectl version`):
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by Fei-Guo** [comment]
Please try call  cloneSetClientset.AppsV1alpha1().CloneSets(lideployment.Namespace).Get() method to see if you can get the created object correctly. 

The clientset APIs are generated by client-go tool, so the returned value of Create() call is determined by how APIServer handles the POST API call (there may be behavior mismatch there). 

Usually, we don't use the return value of Create API directly in the following code. If you want to get the latest  state of the created object, you should call Get() method.

**Comment by wgitg** [comment]
`cloneSetClientset.AppsV1alpha1().CloneSets(foo.Namespace).Get(fooCloneSet, metav1.GetOptions{})` works, Thanks @Fei-Guo 

> so the returned value of Create() call is determined by how APIServer handles the POST API call (there may have behavior mismatch there).

I think it is very possible. FYI, my original foo controller is with kubernetes v0.18.6. (including api-server, api, etc).
the `code-generator` is on v18.6 too. 

But then when I compiled with openkruise/kruise, it seems the modules need to be pined to v0.16.6. So does the code-generator (I need to get the v0.16.6) to be able to generate my crd stuff correctly working. In v0.16.6, the interfaces and APIs generated clientset for CRD foo are changed, not compatible with v0.18.6.

However, for the Kubernetes `Deployment` clientset/lister/informers, it work with both versions (v0.16.6. and v0.18.6). Specifically, the clientset API on Create returns the exact the same deployment object I posted to create. 

So the last question: **can we make openkruise/kruise v0.6.0 or above to work with Kubernetes and its toolchain on v0.18.X above?**  @Fei-Guo
